data_model/src/lib.rs

```diff
@@ -244,23 +250,54 @@ pub enum Parameter {
 )]
 #[allow(clippy::enum_variant_names)]
 pub enum IdBox {
```

Comment:
it cannot be generic. The whole point of having `IdBox` is to have a type which isn't generic

Reply:
the reason why we have these box structures is to be able to reason about structures implementing a certain trait without using trait objects because we cannot serialize/deserialize trait objects. This might be changed with [this](https://github.com/hyperledger/iroha/issues/1983)